applications/HDF5Application/python_scripts/core/file_io.py

```diff
@@ -80,24 +81,47 @@ def _GetIO(io_type):
 class _FilenameGetter(object):
 
     def __init__(self, settings):
-        filename = settings['file_name']
-        self.filename_parts = filename.split('<time>')
+        self.filename = settings['file_name']
+        self.model_part = None
+
+        if (self.filename.find("<rank>") != -1):
+            raise Exception("Flag \"<rank>\" is not allowed to be used in HDF5 output file namings. Please remove it [ filename = \"" + self.filename + "\" ].")
+
+        if (self.filename.find("<step>") != -1):
+            raise Exception("Flag \"<step>\" is not allowed to be used in HDF5 output file namings. Please remove it [ filename = \"" + self.filename + "\" ].")
+
+        if (not self.filename.endswith(".h5")):
+            self.filename += ".h5"
+
+        self.max_files_to_keep = None
+        if ("file_access_mode" in settings.keys() and "max_files_to_keep" in settings.keys()):
```

Comment:
why is related with file_access_mode?

Reply:
Only `truncate` owerrites the files, other modes appends/reads the existing files. So in order to be in safe side, I did not want to delete existing files which may have created from a different simulations (used via `exclusive` or other modes). `truncate` mode always ensures the h5 files is written from scratch so it always anyways deletes previous data in the same file.

Reply:
ok that makes sense